src/tdsops.f90

```diff
@@ -122,6 +123,28 @@ function tdsops_init(n, delta, operation, scheme, n_halo, from_to, &
 
   end function tdsops_init
 
+  pure function get_operation_data_loc(operation, scheme, from_to) result(data_loc)
+  !! Get the output data_loc of a field from its operation and schemes
+    character(*), intent(in) :: operation
+    character(*), intent(in) :: scheme
+    character(*), optional, intent(in) :: from_to
+    integer :: data_loc
+
+    data_loc = NONE
+    !! TODO implement properly
+
+    if (present(from_to)) then
+      select case(from_to)
+      case("v2p")
+        data_loc = CELL
+      case("p2v")
+        data_loc = VERT
+      end select
+    end if
+
+end function
```

Comment:
@semi-h would you be able to write such function?
You have a way better understanding of these things, so I think it is safer if you do it.

The idea is to get the `data_loc` useful for a tdsops, upon allocation of tdsops it then defines `tds_n` with `tds_n = mesh%get_n(dir, data_loc)`.